newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js

```diff
@@ -31,8 +31,29 @@ export const declareExtension = (
     )
     .setExtensionHelpPath(eventsFunctionsExtension.getHelpPath())
     .setIconUrl(eventsFunctionsExtension.getIconUrl());
+
+  declareExtensionDependencies(extension, eventsFunctionsExtension);
 };
 
+/**
+ * Declare the dependencies of an extension from an events based extension.
+ */
+export const declareExtensionDependencies = (
+  extension: gdPlatformExtension,
+  eventsFunctionsExtension: gdEventsFunctionsExtension
+) =>
+  mapVector<gdDependencyMetadata, void>(
+    eventsFunctionsExtension.getAllDependencies(),
+    dependency => {
+      extension
+        .addDependency()
```

Comment:
Here, I was not sure if I should add a method to the platform extension to push the dependency directly on its dependencies vector. I think it was said before that we should avoid exposing too much from the c++, so I decided to go the "verbose way". Is it ok?

Reply:
It's ok to do it in JS because that's how the rest of this file is done.
This being said, it's true that it's verbose and could be even a source of bug if we add a new field later (you add it to the C++ side, you add it in the UI... and it's not working! Then you debug until you find this and ask yourself "why are not all fields copied???").

So I would say:
- The logic of "adding all dependencies" should stay in this JS file (and maybe later we'll rewrite it in C++, but that's not a priority).
- BUT we can still offer a convenient way of copying a DependencyMetadata... it would be easy in C++ (just assign the reference: `auto& extensionDependency = extension.AddDependency(); extensionDependency = dependency`), but not in JS. And we also can't overload `addDependency` 😬

Might be worth adding either a method to copy a dependency (but that's a bit useless in C++...) or a method to copy the entire vector as you mentioned.
Let me know what you think.